install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,408 @@
+
+* CSI driver registrar, which registers the CSI driver into the `openshift-node` service running on the node. The `openshift-node` process running on the node then directly talks to the CSI driver using the Unix Domain Socket available on the node.
+* CSI driver.
+
+The CSI driver deployed on the node should have as few credentials to the storage backend as possible. {product-title} will only use the node plugin set of CSI calls such as `NodePublish`/`NodeUnpublish` and `NodeStage`/`NodeUnstage` (if implemented).
+
+== Example Deployment
+
+Since {product-title} does not ship with any CSI driver installed, this example shows how to deploy a community driver for OpenStack Cinder in {product-title}.
+
+1. Create a new project where the CSI components will run and a new service account that will run the components
+
+[source,shell]
+----
+# oc new-project csi
+Now using project "csi" on server "https://example.com:8443".
+
+# oc create serviceaccount cinder-csi
+serviceaccount "cinder-csi" created
+
+# oc adm policy add-scc-to-user privileged system:serviceaccount:csi:cinder-csi
+scc "privileged" added to: ["system:serviceaccount:csi:cinder-csi"]
+----
+
+2. Apply this yaml file to create DaemonSet with external attacher, Deployment with external provisioner and DaemonSet with the CSI driver.
+
+[source,yaml]
+----
+# This YAML file contains all API objects that are necessary to run Cinder CSI
+# driver.
+#
+# In production, this needs to be in separate files, e.g. service account and
+# role and role binding needs to be created once, while StatefulSet set may
+# require some tuning.
+#
+# It server as an example how to use external attacher and external provisioner
+# images shipped with {product-title} with a community CSI driver.
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cinder-csi-role
+rules:
+  - apiGroups: [""]
+    resources: ["persistentvolumes"]
+    verbs: ["create", "delete", "get", "list", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["volumeattachments"]
+    verbs: ["get", "list", "watch", "update"]
+
+---
+
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cinder-csi-role
+subjects:
+  - kind: ServiceAccount
+    name: cinder-csi
+    namespace: csi
+roleRef:
+  kind: ClusterRole
+  name: cinder-csi-role
+  apiGroup: rbac.authorization.k8s.io
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: cinder-csi
+  labels:
+    app: cinder-csi
+spec:
+  selector:
+    app: cinder-csi
+  ports:
+    - name: dummy
+      port: 12345
+
```

Comment:
Which resource will use the `Service` resource?

Reply:
moved a bit down + renamed to `cinder-csi-attacher` service - StatefulSet requires a service.